fastlane_core/lib/fastlane_core/ui/implementations/shell.rb

```diff
@@ -150,6 +150,17 @@ def password(message)
 
     private
 
+    def encode_as_utf_8_if_possible(message)
+      return message if message.valid_encoding?
+
+      # genstrings outputs UTF-16, so we should try to use this encoding if it turns out to be valid
+      test_message = message.dup
+      return message.encode(Encoding::UTF_8, Encoding::UTF_16) if test_message.force_encoding(Encoding::UTF_16).valid_encoding?
+
+      # replace any invalid with empty string
+      return message.encode(Encoding::UTF_8, invalid: :replace, replace: "")
```

Comment:
Maybe it's worth leaving invalid characters as `�` as people would still want to if there were any chars there?


```suggestion
      return message.encode(Encoding::UTF_8, invalid: :replace)
```

https://ruby-doc.org/core-2.4.0/String.html#method-i-encode